providers/implementations/macs/hmac_prov.c

```diff
@@ -115,6 +115,7 @@ static void *hmac_dup(void *vsrc)
     *dst = *src;
     dst->ctx = ctx;
     dst->key = NULL;
+    memset(&dst->digest, 0, sizeof(dst->digest));
 
     if (!HMAC_CTX_copy(dst->ctx, src->ctx)
         || !ossl_prov_digest_copy(&dst->digest, &src->digest)) {
```

Comment:
Wouldn't it be better to do the memset inside this if instead?

Reply:
this is safer programming, imagine this code pattern will be used elsewhere, and maybe the
ossl_prov_digest_copy is not the last thing in the if condition.